zio-http/src/main/scala/zhttp/service/HttpRuntime.scala

```diff
@@ -20,8 +20,8 @@ final class HttpRuntime[+R](strategy: HttpRuntime.Strategy[R]) {
     rtm
       .unsafeRunAsync(for {
         fiber <- program.fork
-        _     <- ZIO.effectTotal {
-          ctx.channel().closeFuture.addListener((_: AnyRef) => rtm.unsafeRunAsync_(fiber.interrupt): Unit)
+        _     <- ZIO.effect {
```

Comment:
why this change?

Reply:
EffectTotal should be used when we know there won't be any exceptions, but the statement on line 24 can throw.